contracts/Remittance.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity >=0.4.21 <0.6.0;
+
+import "./Pausable.sol";
+
+contract Remittance is Pausable {
+  address public owner;
+  address public carol;
+  bytes32 private withdrawHash;
+
+  event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
+  event LogWithdrawn(address indexed sender, uint256 amount);
+
+  constructor(bool paused) Pausable(paused) public {
+  }
+
+  function calcuateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
```

Comment:
No need of `_`.

Reply:
It is not wrong. What made you include `address carol`?

Reply:
Because of the requirement:
> Carol submits both passwords to Alice's remittance contract.

Bob doesn't care about ether, he may not have an ether account, may not have ether for gas etc. So I think Carol must have an address to receive the fund.

Reply:
@xavierlepretre If I remove `_` from `_carol`, there is a warning says `This declaration shadows an existing declaration.` Because I have a state for the contract called `address public carol;`

Reply:
Ok for shadowing.
About submitting Carol's address, if you did not check for Carol's address, anyone could see the 2 passwords as soon as the tx is sent, even before it was mined. These other people could then create a tx of their own with higher gas price and front-run carol.

Reply:
Since we don't store the passwords and only store `withdrawHash`, how can the others/hack see the passwords?

If the hack doesn't a hack to do a front-run tx to overwrite Carol's address int the second parameter. But they don't know how to generate the same hash to withdraw.  And the hacker's fund (ethers in `msg.value`) will be locked in the contract. The owner's fund didn't get mined, so it's not lost. Am I correct?

Reply:
> Ok for shadowing.

I don't get this part here, why shall we keep a warning here?

Reply:
I meant, I "accept your explanation about shadowing, so I accept your keeping of the _".

About the 2 passwords, I meant at the time of submitting the withdraw transaction. This transaction is not mined immediately and it does contain the 2 passwords. So anyone could see and craft a new tx with them. Unless...
Unless you include Carol's address. Which you did. And I wanted to make sure you understood the importance of this inclusion.